### PR TITLE
Adds types for luxon

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/luxon/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+            weekNumber: number;
+            weekYear: number;
+            weekday: string;
+            weekdayLong: string;
+            weekdayShort: string;
+            year: number;
+            zoneName: string;
+            diff(
+                other: DateTime,
+                unit: string | string[],
```
Comment (sanderpick) [inline types/luxon/index.d.ts]: I thiiiink `unit` [is optional](https://moment.github.io/luxon/docs/class/src/datetime.js~DateTime.html#instance-method-diff).

File: types/luxon/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+                options?: DiffOptions
+            ): Duration;
+            diffNow(unit: string | string[], options?: DiffOptions): Duration;
```
Comment (sanderpick) [inline types/luxon/index.d.ts]: same here

**Comment by sanderpick** [comment]
very cool

**Comment by colbydehart** [comment]
Cool, I updated those types, thanks!

File: types/luxon/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+            numberingSystem: string;
+            offset: number;
+            offsetNameLong: string;
+            offsetNameShort: string;
+            ordinal: number | DateTime;
+            outputCalendar: string;
+            second: number;
+            weekNumber: number;
+            weekYear: number;
+            weekday: string;
```
Comment (FourwingsY) [inline types/luxon/index.d.ts]: weekday should be number, isn't it?

File: types/luxon/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+            numberingSystem?: string;
+        };
+
+        type DateTimeJSOptions = {
+            zone?: string | Zone;
+        };
+
+        type DateObjectUnits = {
+            year?: number;
+            day?: number;
```
Comment (FourwingsY) [inline types/luxon/index.d.ts]: missing month

**Comment by aozgaa** [comment]
Oops, merged the PR from a stale tab, so I didn't see @FourwingsY's comments. Please consider making the changes in a new PR. My apologies for the inconvenience.

**Comment by FourwingsY** [comment]
@aozgaa Oh, I will :)

**Pull request merged**